tests/test_model_generator.py

```diff
@@ -3722,13 +3772,277 @@ def test_modelgen_save_reload_links_03(self):
+
+        # Now refetch the object with all links explicitly
+        billie = self.client.get(default.User.filter(name="Billie"))
+        tsl2 = self.client.get(
+            default.TestSingleLinks.select(
+                "*",
+                req_wprop_friend=True,
+                req_friend=True,
+                opt_friend=True,
+                opt_wprop_friend=True,
+                comp_req_wprop_friend=True,
+                comp_req_friend=True,
+                comp_opt_friend=True,
+                comp_opt_wprop_friend=True,
+            ).filter(id=tsl.id)
+        )
+
+        # Make mypy happy
+        assert tsl2.opt_wprop_friend is not None
+        assert tsl2.opt_friend is not None
+        assert tsl2.comp_opt_friend is not None
+        assert tsl2.comp_opt_wprop_friend is not None
+
+        self.assertEqual(tsl2.comp_req_wprop_friend.name, "Alice")
+        self.assertEqual(tsl2.comp_req_friend.name, "Alice")
+        self.assertEqual(tsl2.comp_opt_friend.name, "Alice")
+        self.assertEqual(tsl2.comp_opt_wprop_friend.name, "Alice")
+        # Check the ids as well
+        self.assertEqual(
+            tsl2.comp_req_wprop_friend.id, tsl.req_wprop_friend.id
+        )
+        self.assertEqual(tsl2.comp_req_friend.id, tsl.req_friend.id)
+        self.assertEqual(tsl2.comp_opt_friend.id, tsl.opt_friend.id)
+        self.assertEqual(
+            tsl2.comp_opt_wprop_friend.id, tsl.opt_wprop_friend.id
+        )
+
+        # Change all links to Billie
+        tsl2.req_wprop_friend = default.TestSingleLinks.req_wprop_friend.link(
+            billie
+        )
+        tsl2.req_friend = billie
+        tsl2.opt_friend = billie
+        tsl2.opt_wprop_friend = default.TestSingleLinks.opt_wprop_friend.link(
+            billie
+        )
+
+        # Save the changes
+        self.client.sync(tsl2)
+
+        # After save, the main links should be updated to Billie
+        self.assertEqual(tsl2.req_wprop_friend.name, "Billie")
+        self.assertEqual(tsl2.req_friend.name, "Billie")
+        self.assertEqual(tsl2.opt_friend.name, "Billie")
+        self.assertEqual(tsl2.opt_wprop_friend.name, "Billie")
+
+        # But computed links should no longer be set after the save
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_req_wprop_friend' is not set"
+        ):
+            assert tsl2.comp_req_wprop_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_req_friend' is not set"
+        ):
+            assert tsl2.comp_req_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_opt_friend' is not set"
+        ):
+            assert tsl2.comp_opt_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_opt_wprop_friend' is not set"
+        ):
+            assert tsl2.comp_opt_wprop_friend is not None  # access field
```

Comment:
This was the result of a `sync(tsl2)`. The object `tsl2` was fetched from the database with computed links being included (so none of them are "unset"). This means that after the sync, we expect the computed links to still be present because they were explicitly fetched originally. So they should have been re-fetched.

Reply:
@msullivan